### PR TITLE
visualize monitor event intervals

### PR DESCRIPTION
**Pull request by deads2k**
Create static e2echart HTML.  Something like this for instance.

Screenshot from 2021-03-30 12-56-00

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/retest

File: pkg/monitor/intervalcreation/operator.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+		from := beginning
+		if lastCondition != nil {
+			from = lastCondition.LastTransitionTime.Time
+		}
+		ret = append(ret, &monitorapi.EventInterval{
+			Condition: &monitorapi.Condition{
+				Level:   level,
+				Locator: event.Locator,
+				Message: fmt.Sprintf("condition/%s status/%s reason/%s", conditionType, lastCondition.Status, lastCondition.Message),
```
Comment (wking) [inline pkg/monitor/intervalcreation/operator.go:67]: /hold

[metal-ipi][1]:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1dc4691]
goroutine 1 [running]:
github.com/openshift/origin/pkg/monitor/intervalcreation.intervalsFromEvents_OperatorStatus(0xc007280000, 0x57af, 0x6000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x5b84c56, ...)
	github.com/openshift/origin/pkg/monitor/intervalcreation/operator.go:67 +0x431
github.com/openshift/origin/pkg/monitor/intervalcreation.IntervalsFromEvents_OperatorProgressing(0xc007280000, 0x57af, 0x6000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x92b6fc8, ...)
	github.com/openshift/origin/pkg/monitor/intervalcreation/operator.go:16 +0xdb
github.com/openshift/origin/pkg/monitor.(*Monitor).EventIntervals(0xc001264a10, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x1, 0x0)
	github.com/openshift/origin/pkg/monitor/monitor.go:141 +0x2f7
...
```

You have a `lastCondition != nil` comparison above, so you need some sort of guard against `lastCondition == nil` here.

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26034/pull-ci-openshift-origin-master-e2e-metal-ipi-ovn-ipv6/1377620127620534272#1:build-log.txt%3A14596

Comment (deads2k) [reply pkg/monitor/intervalcreation/operator.go]: > You have a `lastCondition != nil` comparison above, so you need some sort of guard against `lastCondition == nil` here.

thanks.  Fixed.  Reapplied lgtm and removed hold

**Comment by deads2k** [comment]
simple rebase.  reapplying.

**Comment by deads2k** [comment]
/override ci/prow/e2e-gcp-csi

The test succeeded before and has tests that failed unrelated to this PR

**Comment by wking** [comment]
[e2e-gcp-upgrade][1] is green, and has the rendered chart [here][2].  I love this :)

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26034/pull-ci-openshift-origin-master-e2e-gcp-upgrade/1379041344818778112
[2]: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/26034/pull-ci-openshift-origin-master-e2e-gcp-upgrade/1379041344818778112/artifacts/e2e-gcp-upgrade/openshift-e2e-test/artifacts/e2e-intervals.html

**Pull request merged**